EIPS/eip-1504.md

```diff
@@ -0,0 +1,349 @@
+---
+eip: 1504
+title: ERC-1504 Upgradable Smart Contract
+author: Kaidong Wu <wukd94@pku.edu.cn>, Chuqiao Ren <cr025@bucknell.edu>, Ruthia He <rujiahe@gmail.com>, Yun Ma <mayun@pku.edu.cn>
+discussions-to: https://github.com/ethereum/EIPs/issues/1503
+status: Draft
+type: Standards Track
+category: ERC
+created: 2018-10-17
+---
+
+## Simple Summary
+
+A standard interface/guideline that makes a smart contract upgradable. 
+
+## Abstract
+
+Ethereum smart contracts have suffered a number of security issues in the past few years. The cost of fixing such a bug in smart contract is significant; for example, the consequences of The DAO attack in June 2016 caused tremendous financial loss and the hard fork of Ethereum blockchain.
+
+The following standard makes it possible to upgrade a standard API within smart contracts. This standard provides basic functionalities to upgrade the operations of the contract without data migration. To ensure the decentralization/community interests, it also contains a voting mechanism to control the upgrading process. 
+
+## Motivation
+
+Smart contract is immutable after deployment. If any security risk is identified or program bug is detected, developers always have to destruct the old contract, deploy a new one and potentially migrate the data (hard fork) to the new contract. In some cases, deploying a smart contract with bugs and potential security vulnerabilities can cause a significant amount of financial loss.  
+
+We propose this upgradable contract to fix the current situation. With the upgradable contract, developers can deploy a new version of smart contract after previous deployment and retain the data at the same time. 
+
+For example, after an ERC20-compliant token contract is deployed, the users exploit a vulnerability in the source code.  Without the support of upgradable contract, developers have to fix this issue by deploy a new, secured contract otherwise the attackers would take advantage of the security hole, which may cause a tremendous financial loss. A challenge is how to migrate data from the old contract to a new one. With the upgradable contract below, this will become relatively easy as developers only have to upgrade the Handler contract to fix bugs while the Data contract will remain the same.
+
+## Specification
+
+The upgradable contract consists of three parts:
+
+- **Handler contract** (implements **Handler interface**) defines operations and provides services. This contract can be upgraded;
+- **Data contract** keeps the resources (data) and is controlled by the Handler contract;
+- **Upgrader contract (optional)** deals with the voting mechanism and upgrades the Handler contract. The voters are pre-defined by the contract owner. 
+
+> The following codes are exact copies of the [ERC-54 Upgradable Smart Contract.](https://gist.github.com/swordghost/969648793ed54fdf027cb6df1b894b75)
```

Comment:
There is no such thing as "ERC-54", at least I cannot find such a number in this repo.

Reply:
The mistake has been corrected.